tests/asgi/test_cythonized_asgi.py

```diff
@@ -0,0 +1,82 @@
+import sys
+import time
+
+import pytest
+
+import falcon
+from falcon import testing
+import falcon.asgi
+try:
+    import pyximport
+    pyximport.install()
+except ImportError:
+    pyximport = None
+
+# NOTE(kgriffs): We do this here rather than inside the try block above,
+#   so that we don't mask errors importing _cythonized itself.
+if pyximport:
+    from . import _cythonized  # type: ignore
+
+from _util import disable_asgi_non_coroutine_wrapping  # NOQA
+
+
+@pytest.fixture
+def client():
+    return testing.TestClient(falcon.asgi.App())
+
+
+@pytest.mark.skipif(not pyximport, reason='Cython not installed')
+def test_jsonchema_validator(client):
+    with disable_asgi_non_coroutine_wrapping():
+        client.app.add_route('/', _cythonized.TestResourceWithValidation())
+
+        with pytest.raises(TypeError):
+            client.app.add_route('/wowsuchfail', _cythonized.TestResourceWithValidationNoHint())
+
+    client.simulate_get()
+
+
+@pytest.mark.skipif(not pyximport, reason='Cython not installed')
+def test_scheduled_jobs(client):
+    resource = _cythonized.TestResourceWithScheduledJobs()
+    client.app.add_route('/', resource)
+
+    client.simulate_get()
+    time.sleep(0.5)
+    assert resource.counter['backround:on_get:async'] == 2
+    assert resource.counter['backround:on_get:sync'] == 40
+
+
+@pytest.mark.skipif(not pyximport, reason='Cython not installed')
+@pytest.mark.skipif(
+    sys.version_info.minor < 7,
```

Comment:
This won't work as expected on Python 4.0 :smiling_imp:

Reply:
ah, good point. Might as well future-proof it.